pkg/skaffold/output/output.go

```diff
@@ -101,14 +103,33 @@ func GetUnderlyingWriter(out io.Writer) io.Writer {
 
 // WithEventContext will return a new skaffoldWriter with the given parameters to be used for the event writer.
 // If the passed io.Writer is not a skaffoldWriter, then it is simply returned.
-func WithEventContext(out io.Writer, phase constants.Phase, subtaskID string) (io.Writer, *logrus.Logger) {
+func WithEventContext(out io.Writer, phase constants.Phase, subtaskID string) io.Writer {
 	if sw, isSW := out.(skaffoldWriter); isSW {
 		return skaffoldWriter{
 			MainWriter:  sw.MainWriter,
 			EventWriter: eventV2.NewLogger(phase, subtaskID),
+			task:        phase,
+			subtask:     subtaskID,
 			timestamps:  sw.timestamps,
-		}, nil
+		}
+	}
+
+	return out
+}
+
+// Log takes an io.Writer (ideally of type output.skaffoldWriter) and constructs
+// a logrus.Entry from it, adding fields for task and subtask information
+func Log(out io.Writer) *logrus.Entry {
+	sw, isSW := out.(skaffoldWriter)
+	if isSW {
+		return logrus.WithFields(logrus.Fields{
+			"task":    sw.task,
+			"subtask": sw.subtask,
+		})
 	}
 
-	return out, nil
+	return logrus.WithFields(logrus.Fields{
+		"task":    constants.DevLoop,
```

Comment:
nit: Perhaps add a comment as to why `task: constants.DevLoop` makes sense here

Reply:
done 👍🏼